src/IHaskell/Eval/Evaluate.hs

```diff
@@ -716,20 +729,31 @@ evalCommand output (Expression expr) state = do
     if isWidget
     then registerWidget out
     else return out
-
   else do
-    -- Evaluate this expression as though it's just a statement.
-    -- The output is bound to 'it', so we can then use it.
-    evalOut <- evalCommand output (Statement expr) state
-
-    let out = evalResult evalOut
-        showErr = isShowError out
-
-    -- If evaluation failed, return the failure.  If it was successful, we
-    -- may be able to use the IHaskellDisplay typeclass.
-    return $ if not showErr || useShowErrors state
-            then evalOut
-            else postprocessShowError evalOut
+    if isDeclaration 
```

Comment:
We'll want to switch around the order of these if blocks – I'll explain in the PR discussion briefly.